rust_src/src/windows.rs

```diff
@@ -2017,4 +2017,53 @@ pub fn window_pixel_height_before_size_change(window: LispWindowValidOrSelected)
     window.pixel_height_before_size_change
 }
 
+/// Set width and type of scroll bars of window WINDOW.
+/// WINDOW must be a live window and defaults to the selected one.
+///
+/// Second parameter WIDTH specifies the pixel width for the vertical scroll
+/// bar.  If WIDTH is nil, use the scroll bar width of WINDOW's frame.
+/// Third parameter VERTICAL-TYPE specifies the type of the vertical scroll
+/// bar: left, right, nil or t where nil means to not display a vertical
+/// scroll bar on WINDOW and t means to use WINDOW frame's vertical scroll
+/// bar type.
+///
+/// Fourth parameter HEIGHT specifies the pixel height for the horizontal
+/// scroll bar.  If HEIGHT is nil, use the scroll bar height of WINDOW's
+/// frame.  Fifth parameter HORIZONTAL-TYPE specifies the type of the
+/// horizontal scroll bar: bottom, nil, or t where nil means to not display
+/// a horizontal scroll bar on WINDOW and t means to use WINDOW frame's
+/// horizontal scroll bar type.
+///
+/// Return t if scroll bars were actually changed and nil otherwise.
+#[lisp_fn(
+    name = "set-window-scroll-bars",
+    c_name = "set_window_scroll_bars",
+    min = "1"
+)]
+pub fn set_window_scroll_bars_lisp(
+    window: LispWindowLiveOrSelected,
+    width: LispObject,
+    vertical_type: LispObject,
+    height: LispObject,
+    horizontal_type: LispObject,
+) -> bool {
+    let mut window: LispWindowRef = window.into();
+    let updated_window = unsafe {
```

Comment:
@clample In the original implementation, the window is accessed via `decode_live_window` which checks for selected window being nil first, if yes -> use it instead of the passed window, otherwise check the passed window for being live (otherwise throwing a "wrong type argument exception") and then use it.

Do you think this may be the case here?
reference: window.c line 264

Reply:
@denis631 `LispWindowLiveOrSelected` is meant to encapsulate the logic of `decode_live_window`.

Reply:
Ohhh so the `#lisp_fn` macro does all the necessary transformations? It this case mapping `LispObject` to `LispWindowLiveOrSelected`? I see. Nice. Sorry for false negative

Reply:
@denis631 correct. The `lisp_fn` macro usually works out to be:
```
pub fn new_defun(obj1: LispObject, obj2: LispObject, .......) -> LispObject {
    let result = original_defun(obj1.into(), obj2.into(), ....);
    result.into()
}
```

Which is super simple. All of the error handling, checking, and the like occurs in the `From` implementations. The `lisp_fn` macro gets more complicated when the lisp function is of variable arguments. These collect the args into a Vector.

Our convention is that where a Lisp function takes optional parameters we use the `Option` type. In the C code this is represented with a Nil value. Essentially C's LispObject is really an `Option<LispValue>`.